### PR TITLE
Feature: Allow Single Object Upload Photo/video

### LINKED ISSUE #1127: Allow for a single object upload for posts
**Is your feature request related to a problem? Please describe.**
In the create post modal we have 2 different fields to upload video and image post.

**Describe the solution you'd like**
we want to make  a single field that allows posting both images and videos through it, you can name the field as "Add media".

**Additional context**
make sure you manage the preview of the content within the model itself and manage the close preview state.
Screenshot 2023-12-05 at 9 19 09 PM

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by duplixx**
**What kind of change does this PR introduce?**

 Feature

**Issue Number:**

Fixes #1127 

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**
All the videos and snapshots are present in this pr
 #1193

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

### CODE REVIEW

**Comment by beingnoble03** [comment]
Please fix the failing tests.

**Comment by duplixx** [comment]
image
this workflow error is not part of my pr how should i fix this @palisadoes @beingnoble03

**Comment by palisadoes** [comment]
The introspection test will fail as we have added features to the API that are not yet implemented in Admin

@beingnoble03 
@noman2002 

Please review

File: public/locales/en.json [hunk 1]
```diff
@@ -309,6 +309,7 @@
     "postDetails": "Post Details",
     "postTitle1": "Write title of the post",
     "postTitle": "Title",
+    "addMedia": "Upload Photo or Video",
```
Comment (beingnoble03) [inline public/locales/en.json:312]:
```suggestion
    "addMedia": "Upload Media",
```

Comment (beingnoble03) [reply public/locales/en.json:312]: Add translations to other json files too.

File: src/screens/OrgPost/OrgPost.tsx [hunk 2]
```diff
@@ -382,116 +397,54 @@ function orgPost(): JSX.Element {
...
+                  <video controls data-testid="videoPreview">
+                    <source src={postformState.addMedia} type={file.type} />(
+                    {t('tag')})
+                  </video>
                 )}
-              </>
-            )}
-            {!postformState.postImage && (
-              <>
-                <Form.Label htmlFor="postvideo">{t('video')}</Form.Label>
```
Comment (beingnoble03) [inline src/screens/OrgPost/OrgPost.tsx:442]: I think you can delete `video` from `en.json` and other files too. If it's not in use anywhere else in this component/screen.

File: src/screens/OrgPost/OrgPost.tsx [hunk 3]
```diff
@@ -382,116 +397,54 @@ function orgPost(): JSX.Element {
                 });
               }}
             />
-            {!postformState.postVideo && (
-              <>
-                <Form.Label htmlFor="postPhoto">{t('image')}</Form.Label>
```
Comment (beingnoble03) [inline src/screens/OrgPost/OrgPost.tsx:387]: I think you can delete `image` from `en.json` and other files too. If it's not in use anywhere else in this component/screen.

**Review by beingnoble03 — CHANGES_REQUESTED** [review]
@duplixx please address the comments. Rest lgtm.

**Comment by duplixx** [comment]
@beingnoble03 changes done !!

**Comment by duplixx** [comment]
i have done changes can we merge it @palisadoes @beingnoble03 @noman2002

**Review by noman2002 — APPROVED** [review]
LGTM

**Review by palisadoes — CHANGES_REQUESTED** [review]
Please merge with the latest upstream. 

The tests should pass now

**Review by palisadoes — APPROVED** [review]

**Pull request merged**